plugin/trino-hive/src/test/java/io/trino/plugin/hive/parquet/AbstractTestParquetReader.java

```diff
@@ -1988,6 +1995,85 @@ public void testMapMaxReadBytes()
         tester.testMaxReadBytes(getStandardMapObjectInspector(javaStringObjectInspector, javaLongObjectInspector), values, values, mapType(VARCHAR, BIGINT), maxReadBlockSize);
     }
 
+    @Test
+    public void testBackwardsCompatibleRepeatedStringField()
+            throws Exception
+    {
+        File parquetFile = new File(Resources.getResource("parquet_repeated_primitives/string/").toURI());
+        try (DistributedQueryRunner queryRunner = HiveQueryRunner.builder().build();
+                QueryAssertions assertions = new QueryAssertions(queryRunner)) {
+            queryRunner.execute(format("""
+                        CREATE TABLE table_with_repeated_primitive_string (
+                            myString array<varchar>)
+                        WITH (
+                            external_location = '%s',
+                            format = 'PARQUET')
+                            """,
+                    parquetFile.getAbsolutePath()));
+
+            assertThat(assertions.query("SELECT myString FROM table_with_repeated_primitive_string"))
+                    .result().matches(resultBuilder(queryRunner.getDefaultSession(), new ArrayType(VARCHAR))
+                            .row(Arrays.asList("hello", "world"))
+                            .row(Arrays.asList("good", "bye"))
+                            .row(Arrays.asList("one", "two", "three"))
+                            .build());
+            queryRunner.execute("DROP TABLE table_with_repeated_primitive_string");
```

Comment:
Move to a `finally` block unless this all goes away as a result of using a a page source directly for testing.